maro/cli/inspector/cim_dashboard.py

```diff
@@ -0,0 +1,473 @@
+import math
+import os
+
+import numpy as np
+import pandas as pd
+
+import altair as alt
+import maro.cli.inspector.dashboard_helper as common_helper
+import streamlit as st
+from maro.cli.inspector.params import CIMItemOption
+from maro.cli.inspector.params import GlobalFilePaths as Gfiles
+from maro.cli.inspector.params import GlobalScenarios
+from maro.cli.inspector.visualization_choice import (CIMIntraViewChoice,
+                                                     PanelViewChoice)
+
+
+def start_cim_dashboard(source_path: str, epoch_num: int, prefix: str):
+    """Entrance of cim dashboard.
+
+    Args:
+        source_path (str): Data folder path.
+        epoch_num (int) : Number of data folders.
+        prefix (str): Prefix of data folders.
+    """
+    option = st.sidebar.selectbox(
+        "Data Type",
+        PanelViewChoice._member_names_)
+    if option == PanelViewChoice.Inter_Epoch.name:
+        render_inter_view(source_path, epoch_num)
+    elif option == PanelViewChoice.Intra_Epoch.name:
+        render_intra_view(source_path, epoch_num, prefix)
+    else:
+        pass
+
+
+def render_inter_view(source_path: str, epoch_num: int):
+    """Show CIM summary plot.
+
+    Args:
+        source_path (str): Data folder path.
+        epoch_num (int): Number of snapshot folders.
+    """
+    common_helper.render_h1_title("CIM Inter Epoch Data")
+    sample_ratio = common_helper.get_holder_sample_ratio(epoch_num)
+    # get epoch sample num
+    down_pooling_range = _get_sampled_epoch_range(epoch_num, sample_ratio)
+    option_candidates = CIMItemOption.quick_info + CIMItemOption.port_info + CIMItemOption.booking_info
+    # generate data
+    data = common_helper.read_detail_csv(os.path.join(source_path, Gfiles.ports_sum)).iloc[down_pooling_range]
+    data["remaining_space"] = list(
+        map(lambda x, y, z: x - y - z, data["capacity"], data["full"], data["empty"]))
+    # get formula & selected data
+    filtered_data = common_helper.get_filtered_formula_and_data(GlobalScenarios.CIM, data, option_candidates)
+    data = filtered_data["data"]
+    filtered_option = filtered_data["item_option"]
+    _generate_inter_view_panel(data[filtered_option], down_pooling_range)
+
+
+def render_intra_view(source_path: str, epoch_num: int, prefix: str):
+    """Show CIM detail plot.
+
+    Args:
+        source_path (str): Data folder path.
+        epoch_num (int) : Number of snapshots.
+        prefix (str):  Prefix of data folders.
+    """
+    option_epoch = st.sidebar.select_slider(
+        "Choose an Epoch:",
+        list(range(0, epoch_num)))
+    target_path = os.path.join(source_path, f"{prefix}{option_epoch}")
```

Comment:
I'm a little bit confused about the naming. Here is my understanding:

- source_path: the path of all the logs for a specific experiment.
- {prefix}{option_epoch}: a directory. But I don't know what the ``prefix``

Reply:
Could you add the corresponding docstring to these functions/classes?

Reply:
The understanding of the meaning of the parameters is completely correct. Docstring is added to each function. Did you mean "comments" here?

Reply:
"""One-line-Summary.

Detailes if have.

Args:
    arg1 ...
    arg2 ...

Returns if have